app/models/katello/content_view.rb

```diff
@@ -734,6 +734,28 @@ def related_composite_cvs
       content_views
     end
 
+    def audited_cv_repositories_since_last_publish
+      audited_repositories = Audit.where(auditable_id: self).filter do |a|
+        !a.audited_changes["repository_ids"].nil?
+      end
+      audited_repositories.filter! { |a| a.created_at > latest_version_object.created_at } if latest_version_object
+      audited_repositories
+    end
+
+    def audited_cv_repository_publications_changed
+      audited_repositories_publication = Audit.where(auditable_id: repositories).filter do |a|
+        !a.audited_changes["publication_href"].nil?
+      end
+      audited_repositories_publication.filter! { |a| a.created_at > latest_version_object.created_at } if latest_version_object
+      audited_repositories_publication
+    end
+
+    def needs_publish
```

Comment:
Sorry if I wasn't clear before. The Ruby method should still have a question mark, because it returns a boolean. But the rabl attribute shouldn't.